src/hnsw.c

```diff
@@ -84,10 +176,20 @@ static void hnswcostestimate(PlannerInfo *root,
 #endif
 
     *indexStartupCost = 0;
-    *indexTotalCost = costs.indexTotalCost;
-    *indexSelectivity = costs.indexSelectivity;
+    *indexTotalCost = costs.numIndexPages ? costs.indexTotalCost * (num_blocks_accessed / costs.numIndexPages) : 0;
+    *indexSelectivity = num_tuples_in_index ? costs.numIndexTuples / num_tuples_in_index : 0;
+    // since we try to insert index tuples at the last datablock,
+    // there is no "order" at all that can be assumed.
     *indexCorrelation = 0;
-    *indexPages = costs.numIndexPages;
+    *indexPages = num_blocks_accessed;
+
+    elog(INFO, "Query cost estimator");
```

Comment:
Can you change these to `elog(DEBUG5...` and then do something like [this](https://github.com/lanterndata/lanterndb/blob/5c794a424f72260e1ee87e4dfcb0b49424514b57/test/sql/hnsw_select.sql#L20) in the cost estimate tests in which you would like these logs printed?

Reply:
done